griptape/drivers/vector/qdrant_vector_store_driver.py

```diff
@@ -0,0 +1,212 @@
+
+    def query(
+        self,
+        query: str,
+        count: Optional[int] = None,
+        namespace: Optional[str] = None,
+        include_vectors: bool = False,
+        **kwargs,
+    ) -> list[BaseVectorStoreDriver.Entry]:
+        """
+        Query the Qdrant collection based on a query vector.
+
+        Parameters:
+            query (str): Query string.
+            count (Optional[int]): Optional number of results to return.
+            namespace (Optional[str]): Optional namespace of the vectors.
+            include_vectors (bool): Whether to include vectors in the results.
+
+        Returns:
+            list[BaseVectorStoreDriver.Entry]: List of Entry objects.
+        """
+        query_vector = self.embedding_driver.embed_string(query)
+
+        # Create a search request
+        results = self.client.search(collection_name=self.collection_name, query_vector=query_vector, limit=count)
+
+        # Convert results to QueryResult objects
+        query_results = [
+            BaseVectorStoreDriver.Entry(
+                id=result.id,
+                vector=result.vector if include_vectors else [],
+                score=result.score,
+                meta={k: v for k, v in result.payload.items() if k not in ["_score", "_tensor_facets"]},
+            )
+            for result in results
+        ]
+        return query_results
+
+    def upsert_vector(
+        self,
+        vector: list[float],
+        vector_id: Optional[str] = None,
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        content: Optional[str] = None,
+        **kwargs,
+    ) -> str:
+        """
+        Upsert vectors into the Qdrant collection.
+
+        Parameters:
+            vector (list[float]): The vector to be upserted.
+            vector_id (Optional[str]): Optional vector ID.
+            namespace (Optional[str]): Optional namespace for the vector.
+            meta (Optional[dict]): Optional dictionary containing metadata.
+            content (Optional[str]): The text content to be included in the payload.
+
+        Returns:
+            str: The ID of the upserted vector.
+        """
+
+        if vector_id is None:
+            vector_id = str(uuid.uuid5(uuid.NAMESPACE_DNS, str_to_hash(str(vector))))
+
+        if meta is None:
+            meta = {}
+
+        if content:
+            meta[self.content_payload_key] = content
+
+        points = import_optional_dependency("qdrant_client.http.models").Batch(
+            ids=[vector_id], vectors=[vector], payloads=[meta] if meta else None
+        )
+
+        self.client.upsert(collection_name=self.collection_name, points=points)
+        return vector_id
+
+    def load_entry(self, vector_id: str, namespace: Optional[str] = None) -> Optional[BaseVectorStoreDriver.Entry]:
+        """
+        Load a vector entry from the Qdrant collection based on its ID.
+
+        Parameters:
+            vector_id (str): ID of the vector to load.
```

Comment:
```suggestion
            vector_id (str | int): ID of the vector to load.
```

Reply:
In the BaseVectorStoreDriver class, this parent function has following signature.

@abstractmethod
    def load_entry(self, vector_id: str, namespace: Optional[str] = None) -> Optional[Entry]: ...
    
  I suggest we should not make it to have optional as int.